in_toto/in_toto_sign.py

```diff
@@ -207,33 +234,87 @@ def main():
   arguments it then dumps the corresponding file.
   """
   args = parse_args()
-  rsa_key = in_toto.util.import_rsa_key_from_file(args.key)
+  length = len(args.keys)
+  rsa_key = in_toto.util.import_rsa_key_from_file(args.keys[length-1])
+  FILENAME_FORMAT_LAYOUT = "{file_name}.{keyid:.8}.layout"
+  FILENAME_FORMAT_LINK = "{file_name}.{keyid:.8}.link"
+  index=args.signablepath.rfind('/')
+  source_file_name=args.signablepath[index+1:]
 
   if args.verbose:
     log.logging.getLogger().setLevel(log.logging.INFO)
 
-  try:
-    if args.operator == 'sign':
+  if args.operator == 'sign':
+    try:
       if args.replace_sig:
-        signable_object = replace_sign(args.signablepath, args.key)
-
+        signable_object = replace_sign(args.signablepath, args.keys)
       else:
-        signable_object = add_sign(args.signablepath, args.key)
+        signable_object = add_sign(args.signablepath, args.keys)
 
       if args.infix:
-        signable_object.dump(key=rsa_key)
-        sys.exit(0)
+        if not args.destination:
+          if signable_object._type=='layout':
+            fn = FILENAME_FORMAT_LAYOUT.format(file_name=source_file_name,
+              keyid=rsa_key['keyid'])
+            signable_object.dump(fn)
+            sys.exit(0)
+
+          else:
+            signable_object.dump(key=rsa_key)
+            sys.exit(0)
```

Comment:
@lukpueh Sure. But, what if he specifies, that the keyID must be infixed. True, ```signable_object.dump(key=rsa_key) ``` does not overwrites the file because
1. It takes into account when keyID must be infixed in the file name.
2. Even when it infixes the keyID in the original source file name, it doesn't dump overwrites it since the new file is dumped at a different location than the original source file.

So I guess, the only change that needs to be done, is to dump it at the original location, and overwrite the existing file when -i is not specified, and in the other case, remove the existing file, and dump the one with the infixed keyID in it.  Is it what you mean?